### PR TITLE
impl: kyle delete cards

### LINKED ISSUE #38: Remove Cards Permanently
As a user, I would like to be able to remove a card permanently from my deck after I decide that I don't need it anymore.

Acceptance criteria:
1. There is a delete button associated with the current card
2. No other card other than the current card is removed by selecting "delete"

Tasks:
- [x]  Create a delete button on the edit page
- [x]  Add a delete function
- [x]  If deleting the last card, then make an empty card

### PR DESCRIPTION
**Pull request by emich-kyle**
closes #38

### CODE REVIEW

**Comment by rrstem** [comment]
Please update PR description to match project standards

File: frontend/flashfolio/src/Viewer.js [hunk 1]
```diff
@@ -107,6 +131,8 @@ export default function Viewer({viewMode="view"}) {
 			<button id= "shuf" onClick = {() => shufFunction()}>Shuffle</button> :
 			<button id="unshuf" onClick = {() => shufOn ? unshufFunction() : null}>Unshuffle</button>)
 		}
+		<button onClick={deleteCard}>
```
Comment (rrstem) [inline frontend/flashfolio/src/Viewer.js:134]: Please make sure that button is only visible when `viewMode === "edit"`

We don't want user to be able to delete cards when they're just viewing a deck.

File: frontend/flashfolio/src/Viewer.js [hunk 2]
```diff
@@ -79,19 +79,43 @@ export default function Viewer({viewMode="view"}) {
...
+					setFlashcard(flashdeck.current.Cards[shufOrder[cardIterator]]);
 			}
 			else { setCardIterator(0); }
 		}
 	}, [cardIterator])
 
+	function deleteCard() {
+		/* if there's 1 card, then make an empty card*/
+		if (flashdeck.current.Cards.length === 1) {
+			flashdeck.current.Cards[0] = {};
```
Comment (rrstem) [inline frontend/flashfolio/src/Viewer.js:98]: The object will still need a `FrontSide` value and a `BackSide` value. You can just set these to the empty string.

**Review by rrstem — CHANGES_REQUESTED** [review]
Just a few things to change before merging! Otherwise looks good :+1:

File: frontend/flashfolio/src/Viewer.js (hunk 1 above)
Comment (emich-kyle) [reply frontend/flashfolio/src/Viewer.js]: i have made this change

File: frontend/flashfolio/src/Viewer.js (hunk 2 above)
Comment (emich-kyle) [reply frontend/flashfolio/src/Viewer.js:98]: there is an issue when i try to make .FrontSide and .BackSide empty strings like this: 

```
if (flashdeck.current.Cards.length === 1) {
	flashdeck.current.Cards[0].FrontSide = "";
	flashdeck.current.Cards[0].BackSide = "";
	/* view the blank card */
	setFlashcard(flashdeck.current.Cards[cardIterator]);
	return;
	}
```

with this style, there is a bug that occurs when you delete the last card: the view will not update to the blank card. instead, the view will remain on the last card. however, when the entire card is set to {}, how I've written it originally, the view appears properly blank upon deleting without any errors.

i will look further into how to make this work with empty FrontSide and BackSide attributes

Comment (rrstem) [reply frontend/flashfolio/src/Viewer.js:98]: try

```js
flashdeck.current.Cards[0] = {};
flashdeck.current.Cards[0].FrontSide = "";
flashdeck.current.Cards[0].BackSide = "";
```

Comment (emich-kyle) [reply frontend/flashfolio/src/Viewer.js:98]: I have made the requested change

**Comment by rrstem** [comment]
Please resolve new conflicts

**Review by rrstem — APPROVED** [review]
lgtm :+1:

thanks for making all the changes!

**Pull request merged**